phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/obo/hpo/HpoDiseaseAnnotationParser.java

```diff
@@ -74,44 +78,37 @@
     throw new PhenolRuntimeException("Could not load HPO annotations at " + annotationFile);
   }
 
-  public static Map<TermId, HpoDisease> loadDiseaseMap(String annotationFile, Ontology ontology, List<DiseaseDatabase> databases) {
-    HpoDiseaseAnnotationParser parser = new HpoDiseaseAnnotationParser(annotationFile, ontology, databases);
+  /**
+   * Get a map from HPO terms to diseases. This function includes OMIM, ORPHA, and DECIPHER references.
+   * @param annotationFile path to the the {@code phenotype.hpoa} file
+   * @param ontology reference to HPO Ontology object
+   * @return map with key being an HPO TermId object, and value being a list of TermIds representing diseases.
+   */
+  public static Multimap<TermId, TermId> loadTermToDiseaseMap(Path annotationFile, Ontology ontology, Set<DiseaseDatabase> databasePrefixes) {
+    HpoDiseaseAnnotationParser parser = new HpoDiseaseAnnotationParser(annotationFile, ontology, databasePrefixes);
     try {
-      return parser.parse();
+      parser.parse(); // ignore return value for this
+      return parser.getTermToDiseaseMap();
```

Comment:
I agree and I also think the parser is bigger than necessary. The parser provides two data structures:
- `Map<TermId, HpoDisease> diseaseMap`,
- `ImmutableMultimap<TermId, TermId> phenotypeToDiseaseMap`

The parser does not need to be instantiated and there should be three static `load*` methods for:
- loading `diseaseMap`
- loading `phenotypeToDiseaseMap `, and
- loading & returning both maps in a POJO.

The third method can save a pass through the file in case the user code needs both maps.

I should not have peeked my nose into the `HpoDiseaseAnnotationParser` within this PR. Let me open a ticket for the refactor.